kolibri/core/auth/serializers.py

```diff
@@ -137,6 +138,14 @@ class Meta:
         fields = ("id", "name")
 
 
+class CreateFacilitySerializer(serializers.ModelSerializer):
+    preset = serializers.ChoiceField(choices=facility_presets.choices)
+
+    class Meta:
+        model = Facility
+        fields = ("id", "name", "preset")
```

Comment:
You are taking the preset here, but not doing anything with it. 

See here for an example of how the preset data is used in a serializer in order to set the settings appropriately: https://github.com/learningequality/kolibri/blob/release-v0.16.x/kolibri/core/device/serializers.py#L139

This would require adding a `create` method to this serializer to implement this custom logic.

Reply:
@Jaspreet-singh-1032  thanks for adding this. However, here are a few more comments to further make the code more robust;

- You could set the `preset` and `name` fields as variables instead eg.
  - `preset = validated_data.get("preset")`
- Then add some validation checks to these variables before proceeding to use them
- It's also a good idea to wrap the code around a try..catch block just in case any of the transactions fail